src/api/define.js

```diff
@@ -107,42 +106,22 @@ function prepareForRegistration (name, Ctor) {
   Ctor[$props] = createInitProps(Ctor);
 }
 
-function registerV0Element (name, Ctor) {
-  let res;
-  let uniqueName;
-  try {
-    prepareForRegistration(name, Ctor);
-    res = document.registerElement(name, Ctor);
-  } catch (e) {
-    uniqueName = generateUniqueName(name);
-    prepareForRegistration(uniqueName, Ctor);
-    res = document.registerElement(uniqueName, Ctor);
-  }
-  return res;
-}
-
-function registerV1Element (name, Ctor) {
-  let uniqueName = name;
-  if (window.customElements.get(name)) {
-    uniqueName = generateUniqueName(name);
-  }
-  prepareForRegistration(uniqueName, Ctor);
-  window.customElements.define(uniqueName, Ctor, Ctor.extends ? { extends: Ctor.extends } : null);
-  return Ctor;
-}
-
 export default function (name, opts) {
   if (opts === undefined) {
     throw new Error(`You have to define options to register a component ${name}`);
   }
   const Ctor = typeof opts === 'object' ? Component.extend(opts) : opts;
   formatLinkedAttributes(Ctor);
 
-  if (customElementsV1) {
-    return registerV1Element(name, Ctor);
-  } else if (customElementsV0) {
-    return registerV0Element(name, Ctor);
+  if (!window.customElements) {
```

Comment:
Do you think it's worth keeping here to enable async loading of polyfill / Skate?